specification/search/data-plane/Microsoft.Azure.Search.Service/track1/preview/2019-05-06-preview/examples/SearchServiceCreateIndexer.json

```diff
@@ -34,7 +37,10 @@
           "maxFailedItemsPerBatch": 5
         },
         "fieldMappings": [],
-        "disabled": false
+        "disabled": false,
+        "cache" : {
+          "storageConnectionString": "DefaultEndpointsProtocol=https;AccountName=myStorage;AccountKey=myStorageKey;EndpointSuffix=core.windows.net"
```

Comment:
we do return the connection string as of today. From a quick sync with arvind, it sounds like this is because we also roundtrip other secrets in skillsets/indexers (eg: custom web api urls/headers).

---
In reply to: [349790785](https://github.com/Azure/azure-rest-api-specs/pull/7812#discussion_r349790785) [](ancestors = 349790785)

Reply:
Whether we should/should not is another question - perhaps something we can address in the next API version - but skillset/enrichment related resources always return the json body verbatim unlike data sources/indexes

Reply:
The right long term solution is MSI support. If returning the connection string represents the current state of the API, that's fine -- I just always think of `DataSource` when I see this pattern and didn't realize there's inconsistency here between resources.